### PR TITLE
Improve documentation of some SRAM routines

### PR DESCRIPTION
**Pull request by SatoMew**
The original names for the routines come from these commits: https://github.com/pret/pokeyellow/commit/6faafde364d4f21a2c88d4d60617f98b5d41a5b2 and https://github.com/pret/pokeyellow/commit/4571f8130eab92325d16e94330ae7f92f3ea7caa. They are a bit confusing since Yellow doesn't have an RTC.

This PR also syncs pokeyellow with [pokegold](https://github.com/pret/pokegold/blob/master/home/sram.asm) and [pokecrystal](https://github.com/pret/pokecrystal/blob/master/home/sram.asm) a bit.

### CODE REVIEW

File: engine/movie/oak_speech/oak_speech.asm [hunk 1]
```diff
@@ -232,12 +232,12 @@ IntroDisplayPicCenteredOrUpperRight:
 	ld a, b
 	call UncompressSpriteFromDE
 	ld a, $0
```
Comment (Rangi42) [inline engine/movie/oak_speech/oak_speech.asm:234]:
```suggestion
	ld a, SRAM_DISABLE
```

File: engine/movie/hall_of_fame.asm [hunk 2]
```diff
@@ -201,12 +201,12 @@ HoFLoadPlayerPics:
 	ld a, BANK(RedPicFront)
 	call UncompressSpriteFromDE
 	ld a, $0
```
Comment (Rangi42) [inline engine/movie/hall_of_fame.asm:203]:
```suggestion
	ld a, SRAM_DISABLE
```

File: engine/menus/save.asm [hunk 3]
```diff
@@ -581,7 +581,7 @@ SAVCheckRandomID:
 ; checks if Sav file is the same by checking player's name 1st letter
 ; and the two random numbers generated at game beginning
 ; (which are stored at wPlayerID)s
-	call EnableSRAMAndLatchClockData
+	call EnableSRAM
 	ld a, $01
```
Comment (Rangi42) [inline engine/menus/save.asm:585]:
```suggestion
	ld a, BANK(???)
```

File: engine/menus/save.asm [hunk 4]
```diff
@@ -541,14 +541,14 @@ EmptySRAMBox:
 GetMonCountsForAllBoxes:
 	ld hl, wBoxMonCounts
 	push hl
-	call EnableSRAMAndLatchClockData
+	call EnableSRAM
 	ld a, $2
```
Comment (Rangi42) [inline engine/menus/save.asm:545]:
```suggestion
	ld a, BANK(???)
```

File: engine/menus/save.asm [hunk 5]
```diff
@@ -541,14 +541,14 @@ EmptySRAMBox:
...
 	ld [MBC1SRamBank], a
 	call GetMonCountsForBoxesInBank
 	ld a, $3
```
Comment (Rangi42) [inline engine/menus/save.asm:548]:
```suggestion
	ld a, BANK(???)
```

File: engine/menus/save.asm [hunk 6]
```diff
@@ -500,14 +500,14 @@ BoxNoText:
 EmptyAllSRAMBoxes:
 ; marks all boxes in SRAM as empty (initialisation for the first time the
 ; player changes the box)
-	call EnableSRAMAndLatchClockData
+	call EnableSRAM
 	ld a, 2
 	ld [MBC1SRamBank], a
 	call EmptySRAMBoxesInBank
 	ld a, 3
```
Comment (Rangi42) [inline engine/menus/save.asm:507]: and so on
```suggestion
	ld a, BANK(???)
```

**Comment by aaaaaa123456789** [comment]
```
  ld a, SRAM_BANKING_MODE
  ld [MBC1SRamBankingMode], a
  assert SRAM_BANKING_MODE == BANK("Save Data")
  ld [MBC1SRamBank], a
```

File: engine/movie/hall_of_fame.asm [hunk 7]
```diff
@@ -200,7 +200,7 @@ HoFLoadPlayerPics:
 	ld de, RedPicFront
 	ld a, BANK(RedPicFront)
 	call UncompressSpriteFromDE
-	ld a, $0
+	ld a, SRAM_DISABLE
 	call OpenSRAM
```
Comment (dannye) [inline engine/movie/hall_of_fame.asm]: `OpenSRAM:` takes a bank, not an enable/disable flag. Although this change came from a suggestion on your first commit.. in general it doesn't look like your intention was to "fix" all the calls to `OpenSRAM` in this PR, so you could just revert this if you don't want to fix all of them.

File: engine/movie/oak_speech/oak_speech.asm [hunk 8]
```diff
@@ -231,7 +231,7 @@ IntroDisplayPicCenteredOrUpperRight:
 	push bc
 	ld a, b
 	call UncompressSpriteFromDE
-	ld a, $0
+	ld a, SRAM_DISABLE
 	call OpenSRAM
```
Comment (dannye) [inline engine/movie/oak_speech/oak_speech.asm]: Here too.

**Review by Rangi42 — APPROVED** [review]

**Review by vulcandth — APPROVED** [review]

**Pull request merged**